buffer/src/test/java/io/netty/buffer/SlicedByteBufTest.java

```diff
@@ -227,4 +231,33 @@ public void testWriteIso88591CharSequenceExpand() {
     public void testWriteUtf16CharSequenceExpand() {
         super.testWriteUtf16CharSequenceExpand();
     }
+
+    @Test
+    public void ensureWritableWithEnoughSpaceShouldNotThrow() {
+        ByteBuf buf = Unpooled.buffer(10);
```

Comment:
why not also just use `newBuffer(...)` which will take care of returning slice ?